git/git.go

```diff
@@ -164,11 +164,25 @@ func (t *AnnotatedTag) Verify(keyRings ...string) (string, error) {
 }
 
 // String returns a short string representation of the tag in the format
-// of <name@hash>, for eg: <1.0.0@a0c14dc8580a23f79bc654faa79c4f62b46c2c22>
-func (t *AnnotatedTag) String() string {
+// of <name@hash>, for eg: "1.0.0@a0c14dc8580a23f79bc654faa79c4f62b46c2c22"
+// If the tag is lightweight, it won't have a hash, so it'll simply return
+// the tag name, i.e. "1.0.0".
+func (t *Tag) String() string {
+	if len(t.Hash) == 0 {
+		return t.Name
+	}
 	return fmt.Sprintf("%s@%s", t.Name, t.Hash.String())
 }
 
+// IsAnnotated returns true of if the tag is annotated.
+func (t *Tag) IsAnnotated() bool {
```

Comment:
Any reason why we prefer methods here over functions (versus `IsConcreteCommit`)?

Reply:
i prefer methods, as they make more semantic sense. since you're computing a property of that particular object, i feel a method is more apt than a function.

Reply:
I feel the same, but this now creates a mixed usage pattern within the same library. Which doesn't make the UX intuitive.